app/models/spree/gateway/braintree_gateway_decorator.rb

```diff
@@ -45,10 +47,14 @@ def authorize money, credit_card, options={}
       # Include device data if present
       options[:device_data] = credit_card.device_data
 
+      current_merchant_account_id = BraintreePresenter.new(credit_card.payment_method).merchant_account_id(money)
+      options[:merchant_account_id] = current_merchant_account_id
+      credit_card.payment_method.preferences[:merchant_account_id] = current_merchant_account_id
+
       if (money.to_f / 100) >= credit_card.payment_method.preferred_three_d_threshold
         # Is expensive enough that three_d_secure should be used
-        options[:payment_method_nonce] = true
-        options[:three_d_secure] = {required: true}
+        options[:payment_method_nonce] = credit_card.encrypted_data
+        options[:three_d_secure] = { required: true }
```

Comment:
I think you might be right on changing that from `true` to the value, but it's weird how it has worked so far.

- https://github.com/eric1234/active_merchant/blob/braintree-cse-1.47.0/lib/active_merchant/billing/gateways/braintree_blue.rb#L188

- https://github.com/eric1234/active_merchant/blob/braintree-cse-1.47.0/lib/active_merchant/billing/gateways/braintree_blue.rb#L539-L540

I think this second link lead me to believe that it's a `true/false` value, but maybe not. I'm not 100% sure on the correct value here, but having a string doesn't hurt as it evaluates to true.

Thanks for improving the `{ }` style!